src/content/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/instrument-your-own.mdx

```diff
@@ -0,0 +1,267 @@
+7. On the AWS console form, click the **Add** button to add the layer to your Lambda function.
+8. On your Lambda function's page, with the default **Code** tab selected, scroll down to the **Runtime settings** section and click the **Edit** button.
+9. Make a safe copy of the existing **Handler** value. You'll need it for a later step.
+10. Change the **Handler** value to: `newrelic_lambda_wrapper.handler` and click **Save**.
+11. Switch to the **Configuration** tab on your Lambda function's page.
+12. Select the **Environment variables** sub-tab.
+13. Define the following environment variables:
+     * `NEW_RELIC_ACCOUNT_ID` Set to your New Relic [account ID](/docs/accounts/install-new-relic/account-setup/account-id)
+     * `NEW_RELIC_LAMBDA_HANDLER` Set to your function's original Handler value copied down in step 8 earlier
+    * `NEW_RELIC_LICENSE_KEY` Set to your New Relic license key
+    * `NEW_RELIC_LOG_ENDPOINT` Set to 'https://log-api.newrelic.com/log/v1'
+    * `NEW_RELIC_TELEMETRY_ENDPOINT` Set to 'https://cloud-collector.newrelic.com/aws/lambda/v1'
+14. If you want to send your Lambda function's logs to New Relic, which will include everything your function writes to STDOUT, make sure you set the `NEW_RELIC_EXTENSION_SEND_FUNCTION_LOGS` environment variable to 'true'.
+15. Modify the Execution Role to allow access to the New Relic license key secret.
+    * Find the ARN of the secret named `NEW_RELIC_LICENSE_KEY`.
+    * Add a new inline policy in the function's execution role that looks like this the code below. Replace the `SECRET_ARN` with the value you found above.
+        ```json
+        "Statement": [
+          {
+            "Action": [
+              "secretsmanager:GetSecretValue"
+            ],
+            "Resource": "SECRET_ARN",
+            "Effect": "Allow"
+          }
+        ]
+        ```
+
+<Callout variant="tip">
+  The New Relic layer will automatically deliver the New Relic Ruby agent and be loaded via a Ruby `require` prior to your Lambda function's invocation. To avoid conflicts, do not include a copy of the Ruby agent anywhere else. Feel free to perform any desired New Relic Ruby agent API calls within your function to take advantage of the agent's presence.
+</Callout>
+
+Java, Python, Node.js, and .Net runtimes will follow the same instrumentation flow as Ruby, but with some different handlers: 
+    1. Update your function's handler to point to the newly attached layer in the console for your function:
+        * Java:
+            * `RequestHandler` implementation: `com.newrelic.java.HandlerWrapper::handleRequest`
+            * `RequestStreamHandlerWrapper` implementation: `com.newrelic.java.HandlerWrapper::handleStreamsRequest`
+        * Python and Ruby: `newrelic_lambda_wrapper.handler` (underscores)
+        * Node: `newrelic-lambda-wrapper.handler` (hyphens)
+        * For .Net you don't have to set the handler. 
```

Comment:
Probably no need for a numbered list of one here.

Reply:
Good call.